src/core/icc_colorspace.js

```diff
@@ -152,4 +152,21 @@ class IccColorSpace extends ColorSpace {
   }
 }
 
-export { IccColorSpace };
+class CmykICCBasedCS extends IccColorSpace {
+  static #iccUrl;
+
+  constructor() {
+    const filename = "CGATS001Compat-v2-micro.icc";
+    const xhr = new XMLHttpRequest();
+    xhr.open("GET", `${CmykICCBasedCS.#iccUrl}${filename}`, false);
+    xhr.responseType = "arraybuffer";
+    xhr.send(null);
+    super(new Uint8Array(xhr.response), 4);
```

Comment:
This class will currently have `this.name === "ICCBased"` set, which is "wrong" and will most likely cause bugs since we in various places (e.g. in the `src/core/image.js` file) use the `name`-field to check what type of ColorSpace something is.

Most likely you want something like the following to ensure that this class will still be treated like CMYK as expected.
```suggestion
    xhr.send(null);
    
    super(new Uint8Array(xhr.response), 4);
    this.name = "DeviceCMYK";
```